source/Halibut/Transport/Protocol/WebSocketStream.cs

```diff
@@ -102,9 +114,9 @@ void SendCloseMessage()
             if (context.State != WebSocketState.Open)
                 return;
 
-            var sendCancel = new CancellationTokenSource(TimeSpan.FromSeconds(1));
-            context.SendAsync(new ArraySegment<byte>(new byte[0]), WebSocketMessageType.Close, true, sendCancel.Token)
-                .ConfigureAwait(false).GetAwaiter().GetResult();
+            using(var sendCancel = new CancellationTokenSource(TimeSpan.FromSeconds(1)))
+                context.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, "Closing", sendCancel.Token)
+                    .ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
Not that it probably matters much in what we are doing, but shouldn't the close initiator wait for the close response from the other side?

Reply:
On further reading it looks like `CloseOutputAsync` will sort that out for you

Reply:
It falls into a a heap if I try a `CloseAsync` here, possibly because the connection has already been killed by the other end